server/src/routes/api/v1/votesRouter.js

```diff
@@ -0,0 +1,41 @@
+import express from "express"
+import Vote from "../../../models/Vote.js"
+import Review from "../../../models/Review.js"
+
+const votesRouter = new express.Router()
+
+votesRouter.post("/", async (req, res) => {
+  try{
+    if (req.user) {
+      const userId = parseInt(req.user.id)
+      const reviewId = parseInt(req.body.reviewId)
+      const voteStatus = parseInt(req.body.voteStatus)
+      const currentVote = await Vote.query().findOne({ userId, reviewId })
+      const currentReview = await Review.query().findById(reviewId)
+      let newStatus
+      let voteCount
+      if (!currentVote) {
+        const newVote = await Vote.query().insertAndFetch({
+          userId,
+          reviewId,
+          voteStatus
+        })
+        newStatus = newVote.voteStatus
+      } else {
+        const updatedVote = await Vote.query().patchAndFetchById(
+          currentVote.id,
+          { voteStatus }
+        )
+        newStatus = updatedVote.voteStatus
+      }
+      voteCount = await currentReview.voteCount()
```

Comment:
you can avoid a query if you  just ensure that the voteCount on the frontend is properly updated with up or down, rather than retotalling all of them here. see if you can implement without calling VoteCount() here

Reply:
Similar to your following comment, we wanted to recalculate based on what was persisted in the database rather than assume the count made it thru to the backend.